input/system/crunchy_bridge/client.go

```diff
@@ -0,0 +1,258 @@
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK || len(body) == 0 {
+		return nil, err
+	}
+
+	metricViews := MetricViews{}
+	err = json.Unmarshal(body, &metricViews)
+	if err != nil {
+		return nil, err
+	}
+
+	return &metricViews, nil
+}
+
+func (c *Client) GetCPUMetrics() (*CPUMetrics, error) {
+	metricViews, err := c.getMetrics("cpu")
+	if err != nil {
+		return nil, err
+	}
+
+	metrics := CPUMetrics{}
+	for _, series := range metricViews.Series {
+		switch series.Name {
+		case "cpu_load_iowait":
+			metrics.Iowait = average(series.Points)
+		case "cpu_load_system":
+			metrics.System = average(series.Points)
+		case "cpu_load_user":
+			metrics.User = average(series.Points)
+		case "cpu_load_steal":
+			metrics.Steal = average(series.Points)
+		}
+	}
+
+	return &metrics, err
+}
+
+func (c *Client) GetMemoryMetrics() (*MemoryMetrics, error) {
+	metricViews, err := c.getMetrics("memory")
+	if err != nil {
+		return nil, err
+	}
+
+	metrics := MemoryMetrics{}
+	for _, series := range metricViews.Series {
+		switch series.Name {
+		case "memory_used":
+			metrics.MemoryUsedPct = average(series.Points)
+		case "swap_used":
+			metrics.SwapUsedPct = average(series.Points)
+		}
+	}
+
+	return &metrics, err
+}
+
+func (c *Client) GetIOPSMetrics() (*IOPSMetrics, error) {
+	metricViews, err := c.getMetrics("iops")
+	if err != nil {
+		return nil, err
+	}
+
+	metrics := IOPSMetrics{}
+	for _, series := range metricViews.Series {
+		switch series.Name {
+		case "io_wtps":
+			metrics.Writes = average(series.Points)
+		case "io_rtps":
+			metrics.Reads = average(series.Points)
+		}
+	}
+
+	return &metrics, err
+}
+
+func (c *Client) GetLoadAverageMetrics() (*LoadAverageMetrics, error) {
+	metricViews, err := c.getMetrics("load-average")
+	if err != nil {
+		return nil, err
+	}
+
+	metrics := LoadAverageMetrics{}
+	for _, series := range metricViews.Series {
+		switch series.Name {
+		case "load_average_1":
+			metrics.One = average(series.Points)
+		}
+	}
+
+	return &metrics, err
+}
+
+func (c *Client) GetDiskUsageMetrics() (*DiskUsageMetrics, error) {
+	metricViews, err := c.getMetrics("disk-usage")
+	if err != nil {
+		return nil, err
+	}
+
+	metrics := DiskUsageMetrics{}
+	for _, series := range metricViews.Series {
+		switch series.Name {
+		case "postgres_databases_size_bytes":
+			metrics.DatabaseSize = uint64(average(series.Points))
+		case "postgres_log_size_bytes":
+			metrics.LogSize = uint64(average(series.Points))
+		case "postgres_wal_size_bytes":
+			metrics.WalSize = uint64(average(series.Points))
+		}
+	}
+
+	return &metrics, err
+}
+
+func average(points []MetricPoint) float64 {
+	// With metric-views endpoint, it returns metrics for the last 15 minutes
+	// The latest data point(s) often returns value 0 as there is some lag within the metrics collection on Crunchy side
+	// When calculating average, ignore last 3 data points
```

Comment:
[note for reviewer] Hope this makes sense, but here is some example API response:

```
{
  "name": "disk-usage",
  "series": [
    {
      "is_empty": false,
      "max": null,
      "min": 0,
      "name": "postgres_databases_size_bytes",
      "points": [
        {
          "time": "2023-11-28T07:56:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T07:57:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T07:58:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T07:59:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:00:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:01:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:02:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:03:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:04:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:05:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:06:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:07:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:08:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:09:04Z",
          "value": 30542436
        },
        {
          "time": "2023-11-28T08:10:04Z",
          "value": 0
        }
      ],
      "title": "Size of Postgres databases",
      "unit": "byte"
    },
    ...
  ]
}
```

I picked up 3 as a random number (also I didn't really see more than 3 zeros), happy to tweak if others have some opinions.

Reply:
Hmm. Couldn't we just check if the value is 0, and then ignore it?

Its essentially impossible for a database to have 0 size once its initialized, so that would make it easy to ignore those data points.

Not a fan of using a hard coded number like "ignore the last 3", since that could break if Crunchy changes their API.

Reply:
> Couldn't we just check if the value is 0, and then ignore it?

Uhmm yes and no for this. Definitely yes for the disk size, so good call on that one. However, I've seen that the load average and cpu metrics being zero in the middle (not at the last) of the metrics, so simply ignoring zero is not "accurate" for _all_ metrics.

That said, after giving a thought of pros/cons, I think it's okay to simplify this to ignore zero.
If load average or cpu load is being zero on some data points, I think it's quite safe to assume that the other data points are also close to zero, so calculating average ignoring zero won't change the average that much. We won't have the accurate data even if we do the drop 3 data points way (or, there is also no definition of "accurate" here as we get last 15 mins metrics to show the 10 mins metrics).